fusionSuite/frontend
# new: Manage organizations

**This PR must be merged after https://github.com/fusionSuite/backend/pull/18.**

This PR can be reviewed commit by commit.

Changes proposed in this pull request:

- Add a notification center to display notifications to user (successes and errors)
- Allow to create organizations (name & parent organization)
- Add a page to list the existing organizations
- Allow to delete organizations (with confirmation)
- Display username and its organization in the sidemenu

Few known limitations (that I think are ok for the moment):

- organizations are reloaded each time we go to the list and create pages
- confirmations are handled by `window.confirm()` (i.e. browser-native confirmation dialog)
- permissions of the user are not checked (I don't think it's possible for now)
- it is not possible to update an organization name
- "Organizations" are directly accessible via the sidemenu, but it should move elsewhere in the future
- it is not correctly tested, because it's very difficult with the Angular test suite. I started a different branch to explore how to test with Cypress https://github.com/fusionSuite/frontend/pull/28

List of organizations:

![Capture d’écran 2022-08-17 à 09 34 32](https://user-images.githubusercontent.com/107053378/185061455-c456618e-8d73-4291-883f-84fe6206582f.png)

Creation of an organization:

![Capture d’écran 2022-08-17 à 09 34 40](https://user-images.githubusercontent.com/107053378/185061489-0634b9c1-19b8-4faf-bee4-673a869cca59.png)

How to test the feature manually:

1. make sure that your backend is up-to-date with https://github.com/fusionSuite/backend/pull/18
2. login & click on "Organizations"
3. a first "My organization" should be displayed
4. click on "Create an organization" & create few organizations with different parents
5. go back to "Organizations" and verify that your organizations have been created and are attached to the correct parents
6. delete an organization and verify it's well deleted

Pull request checklist:

- [x] tests are updated → see https://github.com/fusionSuite/frontend/pull/28

_If you think one of the item isn’t applicable to the PR, please check it
anyway and precise `N/A` next to it._

Related to:

- https://github.com/fusionSuite/FusionSuite/issues/41
- https://github.com/fusionSuite/backend/pull/18
- https://github.com/fusionSuite/frontend/pull/28
